### PR TITLE
refactor: switch to REST client server api call

### PR DESCRIPTION
**Pull request by Extheoisah**

### CODE REVIEW

File: client/fresheyes-web/src/app/components/form/proceess-pr.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const { owner, repo, pull_number } = args;
+  const session = await auth();
+  if (!session || !session?.accessToken) {
+    return {
+      error: "You must be logged in to process a pull request",
+      data: null,
+    };
+  }
+
+  const response = await fetch(`${API_URL}/process_pull_request`, {
```
Comment (igbopharaoh) [inline client/fresheyes-web/src/app/components/form/proceess-pr.ts:17]: Non blocking but is this how the path is defined with underscores and not dashes

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+import { processPr } from "./proceess-pr";
 
 const FormSection = () => {
-  const { client } = useGrpcClient();
-
   const [link, setLink] = useState("");
   const [loading, setLoading] = useState({ loader: false, modal: false });
   const [error, setError] = useState("");
-  const [formValues, setFormValues] = useState<{ owner: string; repo: string; pull_number: number }>({
+  const [formValues, setFormValues] = useState<{
```
Comment (igbopharaoh) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx]: I think we can use the `PullRequest` type definition here

**Comment by igbopharaoh** [comment]
Overall great work sir, I just left a few non block comments

File: client/fresheyes-web/src/app/components/form/proceess-pr.ts (hunk 1 above)
Comment (Extheoisah) [reply client/fresheyes-web/src/app/components/form/proceess-pr.ts:17]: Yes. Thats the endpoint from the server

File: client/fresheyes-web/src/app/components/form/FormSection.tsx (hunk 2 above)
Comment (Extheoisah) [reply client/fresheyes-web/src/app/components/form/FormSection.tsx]: addressed here -> https://github.com/Extheoisah/fresheyes-ref-impl/pull/11/commits/e2eb89e3735ae3fce11c967b4dcb9279f6762479

**Pull request merged**